draft-ietf-ppm-dap.md

```diff
@@ -506,17 +504,17 @@ given query type. A query is defined as follows:
 
 ~~~
 enum {
-    next-batch(0),
-    by-batch-id(1),
+    by-batch-id(0),
+    current-batch(1),
 } FixedSizeQueryType;
 
-opaque BatchId[32];
+uint64 BatchId;
```

Comment:
Since other "IDs" have type `opaque [32]`, I wonder if it makes sense to give this a different name. What about `BatchNum`?

Reply:
I like calling it an "ID" because that is an accurate description of this type's semantics -- it's a primary identifier for a batch. The difference in types doesn't bother me so much; IMO, it's fairly natural for IDs for different kinds of entities to have different types, depending on the needs for the ID.

OTOH, we might rename some messages such that a "batch ID" is either an interval (for `time-interval` tasks) or an opaque numeric value (for `fixed-size` tasks), i.e.:

```
struct BatchID {
  select (query_type) {
    case time-interval: Interval batch_interval;
    case fixed-size: BatchID batch_id;
  }
}
```

This would be "cleaner" (IMO) -- this new BatchID struct would effectively replace the Query -- but I'm not sure it's worth taking the time to do right now.

Reply:
Good point about my suggestion. The second idea I like quite a lot as well, though I think we'd have to rename the inner "batch_id" :) If the change is not overly invasive, I say go for it and let's see what it looks like! Just be ready to revert if needed.

Reply:
Actually, thinking about it more, it sounds like what you want is to rename "Query" to "BatchID"?

Reply:
Thinking about this some more, I think I don't want to rename Query to BatchID: while this would work well for `fixed-size`, I think the interval included in `time-interval` isn't quite an ID: specifically, `time-interval` tasks allow queries that span multiple `min_batch_duration` intervals, so for `time-interval` the query is actually describing a range of batches rather than a single batch. IMO, a batch ID would identify a single `min_batch_duration` interval for `time-interval`.

(of course, I could keep Query and separately define BatchID, but at that point the suggested change is no longer a simplification...)